core-bundle/src/EventListener/AdministratorEmailListener.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\Config;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * @internal
+ */
+class AdministratorEmailListener
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(TranslatorInterface $translator, RouterInterface $router, RequestStack $requestStack)
+    {
+        $this->translator = $translator;
+        $this->router = $router;
+        $this->requestStack = $requestStack;
+    }
+
+    public function __invoke(): ?string
+    {
+        if (Config::get('adminEmail')) {
```

Comment:
Does this need an initialized framework here?

Reply:
I'd argue that for hooks you never really need that (as the framework already needs to be running for the hook to be executed).

Reply:
All the other hook listeners in Contao that _do_ inject the `ContaoFramework` also only use it to retrieve an adapter - they do not initialize the framework themselves.

Reply:
Yeah, makes sense. 👍